### PR TITLE
bugfix/rangeselector-unnecessary-redraw

### LINKED ISSUE #20214: `RangeSelector.render` gets called twice on `Chart.redraw`
#### Expected behaviour
When calling `chart.redraw`, each of the elements should only render once, or even 0 times, when it's unnecessary.

#### Actual behaviour
The range selector's render method gets called twice when calling `chart.redraw()`

#### Live demo with steps to reproduce
https://jsfiddle.net/BlackLabel/g7tq01np/27/

#### Product version
v11.2

#### Affected browser(s)
all

### LINKED ISSUE #20970: Range Selector dropdown doesn't open in Gantt chart
#### Expected behaviour
Range Selector dropdown should open upon clicking, displaying the available ranges.

#### Actual behaviour
Range Selector dropdown does not open upon clicking.

#### Live demo with steps to reproduce
https://jsfiddle.net/BlackLabel/yn9hrv3u/

#### Product version
Highcharts Gantt v11.3.0 and newer (used to work correctly in v11.2.0)

#### Affected browser(s)
All

_Bisected to:_ https://github.com/highcharts/highcharts/commit/59f5b17ac90c13e3194fa72c9705f0c4769e930c FYI @pawellysy

### PR DESCRIPTION
**Pull request by pawellysy**
Fixed #20214, `RangeSelector#render` was unnecessarily called 3 times on each chart render.
Also, the range selector does not fire any axis methods anymore.
Also Fixed #20970.

### CODE REVIEW

File: ts/Stock/RangeSelector/RangeSelector.ts [hunk 1]
```diff
@@ -200,6 +199,7 @@ class RangeSelector {
     public rendered?: boolean;
     public selected?: number;
     public zoomText?: SVGElement;
+    public dupa!: string;
```
Comment (bm64) [inline ts/Stock/RangeSelector/RangeSelector.ts:202]: I don't think it's meant to be here 😅

**Review by bm64 — APPROVED** [review]
Just one thing, otherwise looks good.

**Review by kamil-musialowski — APPROVED** [review]
🚀

**Review by pawelfus — CHANGES_REQUESTED** [review]
See visual issues + perf (current one is slower by ~50%?)

File: samples/unit-tests/chart/stockchart-update/demo.js [hunk 2]
```diff
@@ -276,9 +274,7 @@
         });
 
         assert.strictEqual(
-            chart.container.querySelectorAll(
-                'g.highcharts-range-selector-buttons .highcharts-button'
-            ).length,
+            chart.rangeSelector.buttons.length,
             6,
             '6 range selector buttons'
```
Comment (DJTechnoo) [inline samples/unit-tests/chart/stockchart-update/demo.js:279]: Maybe rephrase it like "Range selector should have 6 buttons."

**Review by DJTechnoo — APPROVED** [review]
Just a thing of small significance, looks good!

**Review by pawelfus — APPROVED** [review]
That's a big ✅!

Regarding visual diffs:
- I'm okay with black-and-white demo and arrow placements changes, unless Torstein says otherwise
- Not sure about the timezone one. Looks unrelated, can't recreate locally. But somehow it looks familiar..

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks, great work 👏 

The timezone visual diff is real though. If you open http://localhost:3030/samples/#view/stock/time/individual, you'll see an Error 34 in the console. This doesn't happen on the master.

It can be debugged by putting a debugger statement inside the catch here:

https://github.com/highcharts/highcharts/blob/3c6838c543cb52974f1c6ff5518148ceafd62bd1/ts/Core/Time.ts#L439-L441

I traced it back to `getYTDExtremes` and `updateButtonStates`, where `dataMin` and `dataMax` are undefined, ultimately causing the error when trying to compute timezones for undefined.

**Review by TorsteinHonsi — APPROVED** [review]

**Review by pawelfus — APPROVED** [review]
Thanks!
